br/pkg/restore/client.go

```diff
@@ -282,11 +286,11 @@ func (rc *Client) InitCheckpoint(ctx context.Context, s storage.ExternalStorage,
 			return tree, nil, errors.Trace(err)
 		}
 
+		// The schedulers config is nil, so the restore-schedulers operation is just nil.
+		// Then the undo function would use the result undo of `remove schedulers` operation,
+		// instead of that in checkpoint meta.
 		if meta.SchedulersConfig != nil {
-			undo = mgr.MakeUndoFunctionByConfig(*meta.SchedulersConfig)
-		} else {
-			// the schedulers config is nil, so the restore-schedulers operation is just a nop
-			undo = pdutil.Nop
+			checkpointClusterConfig = meta.SchedulersConfig
```

Comment:
don't overwrite by a nop config from checkpoint meta. Because the current cluster pd scheduler config might not be nop. (modified by user)